### PR TITLE
CRPS for quantiles

### PR DESCRIPTION
**Pull request by simon-hirsch**
Hi, there we go: a draft for the CRPS for quantiles via the Pinball loss. Please have a look!

### CODE REVIEW

File: scoringrules/_crps.py [hunk 1]
```diff
@@ -126,6 +126,48 @@ def twcrps_ensemble(
...
+    observations: "ArrayLike",
+    forecasts: "Array",
+    alpha: "Array",
+    /,
+    axis: int = -1,
+    *,
+    backend: "Backend" = None,
+) -> "Array":
+    r"""Approximate the CRPS from quantile predictions via the Pinball Loss.
+
```
Comment (frazane) [inline scoringrules/_crps.py:139]: could you include a reference and possibly the mathematical formula?

**Review by frazane — COMMENTED** [review]
Only minor details about the current draft. Looks great overall! 

For tests don't bother with something too elaborate - we're going to rewrite the entire test suite soon. In the docs I'd add a short explanation of the quantile-based CRPS.

File: scoringrules/_crps.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (simon-hirsch) [reply scoringrules/_crps.py:139]: Hi, yes sure, done that. We could also add a more generic reference than the energy price forecasting one.

File: scoringrules/core/crps/_gufuncs.py [hunk 2]
```diff
@@ -7,6 +7,25 @@
...
 
 
+@guvectorize(
+    [
+        "void(float32[:], float32[:], float32[:], float32[:])",
+        "void(float64[:], float64[:], float64[:], float64[:])",
+    ],
+    "(),(a),(a)->()",
+)
+def crps_quantile_pinball_gufunc(
```
[the remark this answers is not in the document]
Comment (simon-hirsch) [reply scoringrules/core/crps/_gufuncs.py]: done :+1:

**Comment by simon-hirsch** [comment]
Hi, added some docs and an array size check to enforce that `alpha` has the same length as `forecasts` per observation :)

**Comment by simon-hirsch** [comment]
I've noted that there are actually several formulations of the CRPS via quantiles in the literature. One defines the CRPS as 
$$\text{quantileCRPS} = \frac{2}{|Q|} \sum_{q \in Q} PB_q$$ 
[e.g. here](https://arxiv.org/pdf/2102.00968) and others have
$$\text{quantileCRPS} = \frac{1}{|Q|} \sum_{q \in Q} PB_q$$ 
[e.g. here](https://arxiv.org/pdf/2005.01365) i.e. there is a factor of 1 vs 2. The former is consistent e.g. with `crps_normal`, while the latter seems to be quite common as well. 

I've added the factor of two now to keep it internally consistent. Practically, the factor of two does not change much but I feel like this can be confusing. What are your opinions @frazane, @sallen12 ?

**Comment by frazane** [comment]
Users generally expect the estimated CRPS to be consistent with its closed-form counterpart, so I agree with adding the factor of two.

Could you add a simple test? After this is done and all checks pass, I think we're good to merge :100:

**Comment by simon-hirsch** [comment]
There we go :) It looks like the achievable margin is roughly a percentage error of 1/n_quantiles between the analytical one and the approximation.

**Review by frazane — APPROVED** [review]

**Pull request merged**